site/src/sphinx/client-decorator.rst

```diff
@@ -67,6 +67,113 @@ If your decorator is expected to be reusable, it is recommended to define a new
+        System.err.println("Firstly, executed");
+        ...
+    });
+
+    WebClient myClient = cb.build(WebClient.class);
+
+The following diagram describes how an HTTP request and HTTP response are gone through decorators:
+
+.. uml::
+
+    @startditaa
+    +-----------+  req  +-----------+  req  +-----------+  req  +------------------+    req    +--------------+
+    |           |------>|           |------>|           |------>|                  |---------->|              |
+    | WebClient |       | #1        |       | #2        |       | Armeria          |           |    Server    |
+    |           |  res  | decorator |  res  | decorator |  res  | Networking Layer |    res    |              |
+    |           |<------|           |<------|           |<------|                  |<----------|              |
+    +-----------+       +-----------+       +-----------+       +------------------+           +--------------+
+    @endditaa
+
+If the client is a Thrift client and RPC decorators are inserted, HTTP decorators and RPC decorators are
+separately grouped and executed in reverse order of the insertion:
+
+.. code-block:: java
+
+    ClientBuilder cb = Clients.builder(...);
+
+    // #2 HTTP decorator.
+    cb.decorator((delegate, ctx, httpReq) -> {
+        System.err.println("Fourthly, executed.");
+        ...
+    });
+
+    // #2 RPC decorator.
+    cb.rpcDecorator((delegate, ctx, rpcReq) -> {
+        System.err.println("Secondly, executed.");
+        ...
+    });
+
+    // #1 HTTP decorator.
+    cb.decorator((delegate, ctx, httpReq) -> {
+        System.err.println("Thirdly, executed.");
+        ...
+    });
+
+    // #1 RPC decorator.
+    cb.rpcDecorator((delegate, ctx, rpcReq) -> {
+        System.err.println("Firstly, executed.");
+        ...
+    });
+
+An RPC request is converted into an HTTP request before it's sent to a server.
+So RPC decorators are inserted before the RPC request is converted and HTTP decorators are inserted after the
+request is converted into the HTTP request. The following diagram describes it:
+
+.. uml::
+
+    @startditaa
+    +--------+ req +-----------+ req +-----------+                     +-----------+ req +-----------+
+    |        |---->|           |---->|           |---> RPC to HTTP --->|           |---->|           |----=->
+    | Thrift |     | #1 RPC    |     | #2 RPC    |                     | #1 HTTP   |     | #2 HTTP   |
+    | Client | res | decorator | res | decorator |                     | decorator | res | decorator |
+    |        |<----|           |<----|           |<--- HTTP to RPC <---|           |<----|           |<-=----
+    +--------+     +-----------+     +-----------+                     +-----------+     +-----------+
+    @endditaa
+
+If the decorator modifies the response (e.g. :api:`DecodingClient`) or spawns more requests
+(e.g. :api:`RetryingClient`), the consequence is different depending on the order of the decorators.
```

Comment:
the consequence is ... -> the outcome may be ?

Reply:
Fixed.